tcms/testruns/static/testruns/js/get.js

```diff
@@ -409,9 +409,26 @@ function renderTestExecutionRow(testExecution) {
     // remove from expanded list b/c data may have changed
     delete expandedExecutionIds[expandedExecutionIds.indexOf(testExecution.id)]
 
+    template.find('.change-status-button').click(function () {
+        const statusId = $(this).attr('data-status-id')
+
+        jsonRPC('TestExecution.update', [testExecution.id, {
+            'status': statusId,
+        }], execution => {
+
+            reloadRowFor(execution)
+        })
+
+        const $this = $(this)
+        $this.parents('.list-group-item-container').addClass('hidden')
+        // click the .list-group-item-header, not the .test-execution-element itself, because otherwise the handler will fail
+        $this.parents('.test-execution-element').next().find('.list-group-item-header').click()
```

Comment:
this goes after reloadRowFor() inside the RPC handler, otherwise you get a race condition where the next row is expanded before the current one is reloaded.